File: python/update-AAAA-host.py
#/usr/bin/env python3

"""
     The FreeDns.io project
     This sample updates an AAAA record to the IP address to IPv6 address FE80::0202:B3FF:FE1E:8329
"""

from urllib.parse import urlencode
from urllib import request

url = 'https://freedns.io/request'
params = {
  'username': 'foo',
  'password': '***',
  'host': 'myhost',
  'record': 'AAAA',
  'value': 'FE80::0202:B3FF:FE1E:8329'
}

data = urlencode(params)
req = request.Request(url, data.encode('utf-8'))

try:
  response = request.urlopen(req)
  content = response.read()
  print(content)
except request.URLError as e:
  print("Error: %d %s (%s)" % (e.code, e.reason, e.read()))

File: ruby/update-AAAA-host.rb
#!/usr/bin/ruby

# The FreeDns.io project
# This sample updates an AAAA record to IP address FE80::0202:B3FF:FE1E:8329

require 'net/http'
require 'uri'

data = {
    'username' => 'foo',
    'password' => '***',
    'host' => 'myhost',
    'record' => 'AAAA',
    'value' => 'FE80::0202:B3FF:FE1E:8329'
}

uri = URI('https://freedns.io/request')
res = Net::HTTP.post_form(uri, data)

#res.body will contain the response 
#res.code will contain the return the HTML status code
